src/qumin/representations/utils.py

```diff
@@ -90,8 +93,21 @@ def aggregator(s):
     # Long form
     if set(col_names) < set(paradigms.columns):
         lexemes, cell_col, form_col = col_names
-        paradigms = paradigms.pivot_table(values=form_col, index=lexemes, columns=cell_col,
+
+        # Filter cells before pivoting for speed reasons
+        if cells is not None:
+            par_cols = paradigms[cell_col].unique()
+            todrop = list(set(par_cols)-set(cells))
+            log.info('Dropping rows with following cell values: '+", ".join(sorted(todrop)))
+            for checkname in cells:
```

Comment:
Since we're using sets anyways, something like this is better (the loop with a ValueError on the first thing is less useful than giving the full list at the first run): 

```
unknown_cells = set(cells) - set(par_cols)
if unknown_cells:
    raise ValueError(f"You specified some cells which aren't in the paradigm : {' '.join(unknown_cells)}")
```